spring-grpc-spring-boot-autoconfigure/src/main/java/org/springframework/grpc/autoconfigure/client/GrpcClientProperties.java

```diff
@@ -15,35 +15,415 @@
+import org.springframework.context.EnvironmentAware;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.StandardEnvironment;
+import org.springframework.grpc.client.NegotiationType;
+import org.springframework.grpc.client.VirtualTargets;
+import org.springframework.util.unit.DataSize;
+
+import io.grpc.ManagedChannel;
 
 @ConfigurationProperties(prefix = "spring.grpc.client")
-public class GrpcClientProperties {
+public class GrpcClientProperties implements EnvironmentAware, VirtualTargets {
+
+	/**
+	 * The default channel configuration to use for new channels.
+	 */
+	private final ChannelConfig defaultChannel = new ChannelConfig();
 
-	@NestedConfigurationProperty
-	private final NamedChannel defaultChannel = new NamedChannel();
+	/**
+	 * Map of channels configured by name.
+	 */
+	private final Map<String, ChannelConfig> channels = new HashMap<>();
 
-	private final Map<String, NamedChannel> channels = new HashMap<>();
+	private Environment environment;
 
 	GrpcClientProperties() {
 		this.defaultChannel.setAddress("static://localhost:9090");
+		this.environment = new StandardEnvironment();
 	}
 
-	public Map<String, NamedChannel> getChannels() {
+	public ChannelConfig getDefaultChannel() {
+		return this.defaultChannel;
+	}
+
+	public Map<String, ChannelConfig> getChannels() {
 		return this.channels;
 	}
 
+	@Override
+	public void setEnvironment(Environment environment) {
+		this.environment = environment;
+	}
+
 	/**
-	 * Gets the default {@link NamedChannel} configured for the GRPC client.
-	 * @return the default {@link NamedChannel}
+	 * Gets the configured channel with the given name. If no channel is configured for
+	 * the specified name then one is created using the default channel as a template.
+	 * @param name the name of the channel
+	 * @return the configured channel if found, or a newly created channel using the
+	 * default channel as a template
 	 */
-	public NamedChannel getDefaultChannel() {
-		return this.defaultChannel;
+	public ChannelConfig getChannel(String name) {
+		if ("default".equals(name)) {
+			return this.defaultChannel;
+		}
+		ChannelConfig channel = this.channels.get(name);
+		if (channel != null) {
+			return channel;
+		}
+		channel = this.defaultChannel.copy();
+		String address = name;
+		if (!name.contains(":/") && !name.startsWith("unix:")) {
+			address = "static://" + name;
+		}
+		channel.setAddress(address);
+		return channel;
+	}
+
+	@Override
+	public String getTarget(String authority) {
+		ChannelConfig channel = this.getChannel(authority);
+		String address = channel.getAddress();
+		if (address.startsWith("static:") || address.startsWith("tcp:")) {
+			address = address.substring(address.indexOf(":") + 1).replaceFirst("/*", "");
+		}
+		return this.environment.resolvePlaceholders(address);
+	}
+
+	/**
+	 * Represents the configuration for a {@link ManagedChannel gRPC channel}.
+	 */
+	public static class ChannelConfig {
```

Comment:
This is identically the NamedChannel from core renamed to ChannelConfig. Removing the "named" from the name removes the implication that it is has a name but rather it is just channel configuration. 

I am ok w/ `NamedChannel` as well though and really I know better than renaming during a move around like this. I am glad to move it back - just lmk.